packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx

```diff
@@ -212,10 +212,15 @@ export class SearchInWorkspaceResultTreeWidget extends TreeWidget {
         this.cancelIndicator = new CancellationTokenSource();
         const cancelIndicator = this.cancelIndicator;
         const token = this.cancelIndicator.token;
+        const progress = await this.progressService.showProgress({ text: `search: ${searchTerm}`, options: { location: 'search' } });
         token.onCancellationRequested(() => {
             this.changeEmitter.fire(this.resultTree);
+            progress.cancel();
+            if (searchId) {
+                this.searchService.cancel(searchId);
+            }
+            this.cancelIndicator = undefined;
```

Comment:
[minor]: I think `this.changeEmitter.fire(this.resultTree);` should come after this line. When everything has been canceled and unset, you can fire the event. What do you think?

Reply:
done